storage/innobase/include/fsp0sysspace.h

```diff
@@ -282,6 +282,17 @@ extern SysTablespace srv_sys_space;
 /** The control info of a temporary table shared tablespace. */
 extern SysTablespace srv_tmp_space;
 
+/** Check if the space_id is for a shared system tablespace.
+@param[in]	id	Space ID to check
+@return true if id is a system tablespace, false if not. */
+UNIV_INLINE
+bool
+is_shared_system_tablespace(
+	ulint	id)
+{
+	return(id == srv_sys_space.space_id());
```

Comment:
In theory the implementation should go fsp0sysspace.ic file, but I don't think we care about UNIV_NONINL (?) build

Reply:
upstream `is_system_tablespace` and `is_system_or_undo_tablespace` are both in .h file